src/colvarmodule.cpp

```diff
@@ -1822,9 +1823,15 @@ std::istream & colvarmodule::read_objects_state(std::istream &is)
     }
 
     if (is.tellg() == pos) {
-      // This block has not been read by any object: discard it and move on
-      // to the next one
-      is >> colvarparse::read_block(word, NULL);
+      // This block has not been read by any object: discard it , print warning and move on
+      std::string conf;
+      is >> colvarparse::read_block(word, &conf);
+      std::string name;
+      if (parse_restart.key_lookup(conf, "name", &name)) {
```

Comment:
`key_lookup()` modifies internal data: are you okay if we declare `parse_restart` right in this block?